emop-maven-plugin/src/main/java/edu/cornell/MonitorMethodsMojo.java

```diff
@@ -0,0 +1,189 @@
+                    + File.separator + "javamop-agent"
+                    + File.separator + "1.0"
+                    + File.separator + "javamop-agent-1.0.jar";
+        }
+        Util.replaceFileInJar(javamopAgent, "/META-INF/aop-ajc.xml", getArtifactsDir() + File.separator + MONITOR_FILE);
+        long end = System.currentTimeMillis();
+        getLog().info("[eMOP Timer] Generating aop-ajc.xml and replace it takes " + (end - start) + " ms");
+    }
+
+    /**
+     * Generates the set of package names of classes that should be monitored. If the set is non-empty, then any
+     * package not included in this set will not be monitored.
+     * @return created set of package names for weaving. An empty set is returned if the includeLibraries is true.
+     */
+    private Set<String> retrieveIncludePackages() {
+        if (!includeLibraries) {
+            return Util.retrieveProjectPackageNames(getClassesDirectory());
+        }
+        return new HashSet<>();
+    }
+
+    /**
+     * Generates a String containing !within() pointcuts so that instrumentation is only performed within the
+     * affected classes, effectively disabling instrumentation in non-affected classes.
+     * If the includeNonAffected parameter is set to true, then this method would return an empty string.
+     */
+    private String generateNonAffectedExclusion() {
+        // StringBuilder stringBuilder = new StringBuilder();
+        // if (!includeNonAffected) {
+        //     for (String className : getNonAffected()) {
+        //         if (!className.contains("package-info")) { // not excluding package-info results in compile error
+        //             stringBuilder.append("    !within(" + className + ") &&\n");
+        //         }
+        //     }
+        // }
+        // return stringBuilder.toString();
+        return null;
+    }
+
+    // Copied from STARTS
+    private boolean isSameClassPath(List<String> sfPathString) throws MojoExecutionException {
+        if (sfPathString.isEmpty()) {
+            return true;
+        }
+        String oldSfPathFileName = Paths.get(getArtifactsDir(), SF_CLASSPATH).toString();
+        if (!new File(oldSfPathFileName).exists()) {
+            return false;
+        }
+        try {
+            List<String> oldClassPathLines = Files.readAllLines(Paths.get(oldSfPathFileName));
+            if (oldClassPathLines.size() != 1) {
+                throw new MojoExecutionException(SF_CLASSPATH + " is corrupt! Expected only 1 line.");
+            }
+            List<String> oldClassPathelements = getCleanClassPath(oldClassPathLines.get(0));
+            // comparing lists and not sets in case order changes
+            if (sfPathString.equals(oldClassPathelements)) {
+                return true;
+            }
+        } catch (IOException ioe) {
+            ioe.printStackTrace();
+        }
+        return false;
+    }
+
+    // Copied from STARTS
```

Comment:
extend comment to say what the method does; we should not expect familiarity with STARTS